include/posix/pthread_key.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __POSIX_THREAD_KEY_H__
+#define __POSIX_THREAD_KEY_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef CONFIG_PTHREAD_IPC
+#include <misc/dlist.h>
+#include <zephyr/types.h>
+
+typedef u32_t pthread_once_t;
+
+/* pthread_key */
+typedef void *pthread_key_t;
+
+typedef struct pthread_key_obj {
+	/* List of pthread_key_data objects that contain thread
+	 * specific data for the key
+	 */
+	sys_dlist_t key_data_l;
```

Comment:
Same comment: this could be an slist and save a word of storage in every struct.